### ISSUE TITLE
mvn clean install failure for sdk-example

### ISSUE THREAD

**Issue by cicoyle** [dapr/java-sdk#997]
## Expected Behavior

I would like to run `./mvnw clean install` successfully and run the examples successfully. Specifically, I need the `dapr-sdk-examples` jar file, but that is failing.

## Actual Behavior

I see errors in my IDE even on the master branch:
![image](https://github.com/dapr/java-sdk/assets/25733780/1c480a26-05c6-4792-8a55-8197d5d92df3)

Additionally, I see:
```
# from root run
$ ./mvnw clean install
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] dapr-sdk-parent 1.11.0-SNAPSHOT .................... SUCCESS [  1.419 s]
[INFO] dapr-sdk-autogen 1.11.0-SNAPSHOT ................... SUCCESS [  1.319 s]
[INFO] dapr-sdk 1.11.0-SNAPSHOT ........................... SUCCESS [ 18.422 s]
[INFO] dapr-sdk-actors 1.11.0-SNAPSHOT .................... SUCCESS [  7.121 s]
[INFO] dapr-sdk-workflows 0.11.0-SNAPSHOT ................. SUCCESS [  5.069 s]
[INFO] dapr-sdk-springboot 1.11.0-SNAPSHOT ................ SUCCESS [  2.841 s]
[INFO] dapr-sdk-springboot3 1.11.0-SNAPSHOT ............... SUCCESS [  0.438 s]
[INFO] dapr-sdk-examples 1.11.0-SNAPSHOT .................. FAILURE [  0.414 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  37.458 s
[INFO] Finished at: 2024-01-29T17:22:17-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project dapr-sdk-examples: Fatal error compiling: error: release version 17 not supported -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :dapr-sdk-examples

```

## Steps to Reproduce the Problem

Check out master. Run `./mvnw clean install` from root. See errors.

**Comment by cicoyle** [comment]
@skyao Do I need to do something to get the dapr-sdk-examples to be successful? I tried switching the version to be 11 manually in the sdk-examples pom.xml and that resulted in even worse errors. I know [this PR adding jdk17 support](https://github.com/dapr/java-sdk/pull/975/files#diff-a5c0f08bc43f53a0451cad4a4a2e435ce55ad34183350dfae5e4b2c7dca6360b) got merged recently, but I was unaware of it resulting in failures. Thoughts?
